src/transformers/modeling_utils.py

```diff
@@ -1441,7 +1444,7 @@ def _from_config(cls, config, **kwargs):
             torch_dtype (`torch.dtype`, *optional*):
                 Override the default `torch.dtype` and load the model under this dtype.
         """
-        torch_dtype = kwargs.pop("torch_dtype", None)
+        torch_dtype = kwargs.pop("torch_dtype", torch.get_default_dtype())
```

Comment:
because of the above comment about FA2 warnings that model dtype should be half-precision. When we call `_from_config` within modeling code, we don't pass `dtype`. In general it doesn't break anything and the correct dtype is used, but dispatching attention implementation checks for this particular kwarg from inputs. So when the kwarg is None, it assumed we are not in half-precision and raises a warning that "FA2 behavior might be bad and unexpected"

Reply:
let's add a small comment at the top!

Reply:
that might go in a different PR from contributor, the one you linked to this PR